src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -307,11 +307,11 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                 }
             }
 
-            private void FileSystemEventCallback( 
+            private void FileSystemEventCallback(
                 FSEventStreamRef streamRef, 
                 IntPtr clientCallBackInfo, 
                 size_t numEvents, 
-                String[] eventPaths, 
+                String[] eventPaths,
```

Comment:
There isn't much of a win here until you change the imports to stop allocating strings. The change as it stands is a small step towards actually addressing the work item, which is to convert UTF-8 into a buffer and not allocate a managed string until it has passed our filters.